src/components/media-player/timecode-fine-tunning-input.js

```diff
@@ -1,81 +1,92 @@
 import PropTypes from 'prop-types';
-import classNames from 'classnames';
-import { Button, Select } from 'antd';
+import { Button, Input } from 'antd';
 import { useTranslation } from 'react-i18next';
-import React, { useEffect, useState } from 'react';
-import { MinusOutlined, PlusOutlined } from '@ant-design/icons';
-import { formatMillisecondsAsDuration } from '../../utils/media-utils.js';
+import EditIcon from '../icons/general/edit-icon.js';
+import SaveIcon from '../icons/general/save-icon.js';
+import CloseIcon from '../icons/general/close-icon.js';
+import React, { Fragment, useEffect, useState } from 'react';
+import { formatMillisecondsAsDuration, tryConvertDurationToMilliseconds } from '../../utils/media-utils.js';
 
-const FINE_TUNNING_STEPS_IN_MS = [100, 500, 1000];
+const getDuration = milliseconds => {
+  return formatMillisecondsAsDuration(milliseconds, { millisecondsLength: 3 });
+};
 
 function TimecodeFineTunningInput({
   value,
   lowerLimit,
   upperLimit,
-  roundToLowerLimit,
-  roundToUpperLimit,
   disabled,
   onValueChange
 }) {
   const { t } = useTranslation('timecodeFineTunningInput');
-  const [isPlusDisabled, setIsPlusDisabled] = useState(false);
-  const [isMinusDisabled, setIsMinusDisabled] = useState(false);
-  const [fineTunningStep, setFineTuningStep] = useState(FINE_TUNNING_STEPS_IN_MS[0]);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [isInvalid, setIsInvalid] = useState(false);
+  const [valueAsDuration, setValueAsDuration] = useState(null);
 
   useEffect(() => {
-    if (roundToLowerLimit) {
-      setIsMinusDisabled(value === lowerLimit);
-    } else {
-      const nextMinusValue = value - fineTunningStep;
-      setIsMinusDisabled(nextMinusValue < lowerLimit);
-    }
+    setValueAsDuration(getDuration(value));
```

Comment:
We have to call `setIsInvalid` here too, unless it stays visible even if the value is OK (testable for example when typing an invalid value and then clicking with the mouse on a different part in the timeline, then tha value changes, set from the outside, but the invalid state stays true).

Reply:
Good find! 😳 I had not tested this scenario with the timeline, I tested the validation cases with the timerange only.
Fixed.